### PR TITLE
feat: enable single- and multiple-choice questions in the inquiry preview

### PR DESCRIPTION
**Pull request by StephanH90**
## Description

This enables the inquiry preview for single and multiple choice questions:

image

### CODE REVIEW

File: packages/testing/addon/mirage-graphql/index.js [hunk 1]
```diff
@@ -45,11 +45,14 @@ export default function createGraphqlHandler(server) {
...
-            __typename: "SelectedOption",
-          };
+          if (option) {
+            return {
+              slug: value,
+              label: option.label,
+              __typename: "SelectedOption",
+            };
+          }
+          return null;
```
Comment (czosel) [inline packages/testing/addon/mirage-graphql/index.js:55]: cc @anehx

**Comment by czosel** [comment]
Otherwise LGTM :+1:

File: packages/distribution/addon/components/cd-inquiry-dialog/inquiry-part.js [hunk 2]
```diff
@@ -68,6 +73,22 @@ export default class CdInquiryDialogInquiryPartComponent extends Component {
           .map((edge) => ({
             question: edge.node.question.label,
             value: edge.node.value,
+            selectedOption: edge.node?.selectedOption,
+            selectedOptions: edge.node?.selectedOptions,
```
Comment (anehx) [inline packages/distribution/addon/components/cd-inquiry-dialog/inquiry-part.js:77]: Why the optional chaining after `node`?

File: packages/distribution/addon/components/cd-inquiry-dialog/inquiry-part.js [hunk 3]
```diff
@@ -68,6 +73,22 @@ export default class CdInquiryDialogInquiryPartComponent extends Component {
...
+            value: answer.selectedOption
+              ? answer.selectedOption.label
+              : answer.value,
+          }))
+          // For multiple choice the value is a list of selectedOption labels
+          .map((answer) => ({
+            ...answer,
+            value: answer.selectedOptions
+              ? answer.selectedOptions.edges.map((edge) => edge.node.label)
+              : answer.value,
```
Comment (anehx) [inline packages/distribution/addon/components/cd-inquiry-dialog/inquiry-part.js:91]: I think we should put this into a function and then use it in the first `.map()`.

File: packages/distribution/addon/components/cd-truncated.hbs [hunk 4]
```diff
@@ -1,8 +1,16 @@
...
-  {{~this.displayedText~}}
-  {{#if this.truncate}}
-    <a href="" {{on "click" this.toggleExpand}}>
-      {{~t (concat "caluma.distribution." (if this.expand "less" "more"))~}}
-    </a>
-  {{/if}}
-</p>
\ No newline at end of file
+{{#if (is-array @text)}}
+  <UkList @bullet={{true}} class="uk-margin-remove" as |list|>
```
Comment (anehx) [inline packages/distribution/addon/components/cd-truncated.hbs:2]: Don't forget `...attributes`

File: packages/testing/addon/mirage-graphql/index.js (hunk 1 above)
Comment (anehx) [reply packages/testing/addon/mirage-graphql/index.js:55]: Makes sense. I'd put the null return into the if as that should be the "special case" but that's just personal flavour.

**Review by anehx — CHANGES_REQUESTED** [review]

**Review by anehx — APPROVED** [review]

**Comment by anehx** [comment]
LGTM, make sure to squash the commits so we don't have `fix: mr fixes/changes` in the changelog..

**Pull request merged**